homeassistant/components/device_tracker/__init__.py

```diff
@@ -420,18 +423,21 @@ def async_seen(self, host_name: str=None, location_name: str=None,
         self.last_seen = dt_util.utcnow()
         self.host_name = host_name
         self.location_name = location_name
-        self.gps_accuracy = gps_accuracy or 0
+
         if battery:
             self.battery = battery
         if attributes:
             self._attributes.update(attributes)
 
-        self.gps = None
-
         if gps is not None:
             try:
                 self.gps = float(gps[0]), float(gps[1])
+                self.gps_accuracy = gps_accuracy or 0
+                self.gps_updated = dt_util.utcnow()
```

Comment:
If you rather set last_updated to self.last seen, you can compare these two later in `async_update` to see when you have a non-gps dev tracker updating